### PR TITLE
feat(olog): add support for SetOutput

### PR DESCRIPTION
**Pull request by yj-yan**
Please read [CONTRIBUTING.md](CONTRIBUTING.md) for additional information on contributing to this repository!

## What this PR does / why we need it
The function is to change the default output to a given writer, which changes the logs output of the module. 

## Jira ID

[DT-4416]

## Notes for your reviewers

[DT-4416]: https://outreach-io.atlassian.net/browse/DT-4416?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

File: pkg/olog/olog_test.go [hunk 1]
```diff
@@ -113,3 +115,35 @@ func TestLogWithAppInfoHook(t *testing.T) {
...
+	// Force JSON handler for valid unmarshaling used in the TestCapturer
+	SetDefaultHandler(JSONHandler)
+
+	logCapture := NewTestCapturer(t)
+	// Initialize test app info
+	app.SetName("ologHooksTest")
+
+	f, err := os.CreateTemp(".", "test_logfile.log")
+	if err != nil {
+		panic(err)
```
Comment (malept) [inline pkg/olog/olog_test.go:129]: suggestion: avoid panics
```suggestion
		t.Fatalf("Could not create temp file: %v", err)
```

File: pkg/olog/default_handler.go [hunk 2]
```diff
@@ -25,7 +25,7 @@ var (
 	defaultHandler atomic.Int32
 
 	// defaultOut is the default output for the default handler. This is
-	// set to `os.Stderr` by default.
+	// set to `os.Stderr` by default. Can be update by SetOutput()
```
Comment (malept) [inline pkg/olog/default_handler.go:28]: Nit: docs formatting

```suggestion
	// set to `os.Stderr` by default. Can be updated by [SetOutput].
```

File: pkg/olog/olog_test.go [hunk 3]
```diff
@@ -113,3 +115,35 @@ func TestLogWithAppInfoHook(t *testing.T) {
...
+
+func TestOutputLog(t *testing.T) {
+	// Force JSON handler for valid unmarshaling used in the TestCapturer
+	SetDefaultHandler(JSONHandler)
+
+	logCapture := NewTestCapturer(t)
+	// Initialize test app info
+	app.SetName("ologHooksTest")
+
+	f, err := os.CreateTemp(".", "test_logfile.log")
```
Comment (malept) [inline pkg/olog/olog_test.go:127]: question: why is the `dir` parameter the current directory and not the system temporary directory?

File: pkg/olog/olog.go [hunk 4]
```diff
@@ -115,6 +116,11 @@ func NewWithHandler(h slog.Handler) *slog.Logger {
 	return slog.New(h)
 }
 
+// SetOutput set the global output to desired writer.
```
Comment (malept) [inline pkg/olog/olog.go:119]: suggestion: wording

```suggestion
// SetOutput sets the global logger output to desired writer.
```

**Review by malept — CHANGES_REQUESTED** [review]

File: pkg/olog/olog_test.go (hunk 3 above)
Comment (yj-yan) [reply pkg/olog/olog_test.go:127]: Good catch. I was saving the file to local dir to view the content. I will update it to `/tmp` for the temp file.

**Review by malept — APPROVED** [review]

File: pkg/olog/olog_test.go [hunk 5]
```diff
@@ -113,3 +115,37 @@ func TestLogWithAppInfoHook(t *testing.T) {
...
+
+func TestOutputLog(t *testing.T) {
+	// Force JSON handler for valid unmarshaling used in the TestCapturer
+	SetDefaultHandler(JSONHandler)
+
+	logCapture := NewTestCapturer(t)
+	// Initialize test app info
+	app.SetName("ologHooksTest")
+
+	f, err := os.CreateTemp("/tmp", "test_logfile.log")
```
Comment (jaredallard) [inline pkg/olog/olog_test.go:127]: For the future, you can use [`t.TempDir()`](https://pkg.go.dev/testing#T.TempDir) to get a test-specific temporary directory that's automatically deleted at the end of the test run.

File: pkg/olog/olog.go [hunk 6]
```diff
@@ -115,6 +116,11 @@ func NewWithHandler(h slog.Handler) *slog.Logger {
 	return slog.New(h)
 }
 
+// SetOutput sets the global logger output to desired writer.
+func SetOutput(w io.Writer) {
+	defaultOut = w
+}
```
Comment (jaredallard) [inline pkg/olog/olog.go:122]: This isn't thread-safe. You'll probably want to call this out in the function signature, or have a lock protecting `defaultOut`.

**Pull request merged**